examples/basics/multi_input_example.exs

```diff
@@ -18,8 +20,8 @@ defmodule XOR do
   end
 
   defp batch do
-    x1 = Nx.tensor(for _ <- 1..32, do: [Enum.random(0..1)])
-    x2 = Nx.tensor(for _ <- 1..32, do: [Enum.random(0..1)])
+    x1 = Nx.tensor(for _ <- 1..@batch_size, do: [Enum.random(0..1)])
+    x2 = Nx.tensor(for _ <- 1..@batch_size, do: [Enum.random(0..1)])
```

Comment:
```suggestion
    {x1, new_key} = Nx.Random.randint(key(), 0, 1 + 1, shape: {@batch_size})
    {x2, _new_key} = Nx.Random.randint(new_key, 0, 1 + 1, shape: {@batch_size})
```

Reply:
Thanks for the suggestion.

Just realized after applying it and testing it locally that as is it generates always the same batches because of the batch always starting with the same key for every batch.

In order to make it work really randomly we need to keep the returned new keys feed back into the batch generation while streaming data I guess.

Reply:
Yes, you're right